xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/ConnectionStateModel.java

```diff
@@ -0,0 +1,35 @@
+package info.bitrich.xchangestream.service.netty;
+
+import io.reactivex.Observable;
+import io.reactivex.subjects.BehaviorSubject;
+import io.reactivex.subjects.Subject;
+
+/**
+ * The simplest Connection state model possible (maybe someone wants to add authenticated,
+ * connecting, etc...)
+ */
+public final class ConnectionStateModel {
+
+  public enum State {
+    closed,
+    open
+  }
+
+  private volatile State state = State.closed;  // start with a closed state
```

Comment:
would this make more sense as an atomic reference? I would imagine if more states are introduced you may want updates to this to happen atomically no?

Reply:
I think volatile achieves the same thing as Atomic, in this case, but maybe i can use Atomic.getAndSet for the old value